src/main/java/com/autotune/analyzer/autoscaler/instaslice/InstasliceHelper.java

```diff
@@ -0,0 +1,137 @@
+
+        List<GenericKubernetesResource> instasliceObjects = getInstasliceObjects();
+        ObjectMapper mapper = new ObjectMapper();
+
+        if (null != instasliceObjects && !instasliceObjects.isEmpty()) {
+            try {
+                instasliceObjects.forEach(item -> {
+                    String name = item.getMetadata().getName();
+                    LOGGER.debug(AnalyzerConstants.AutoscalerConstants.InfoMsgs.FOUND_INSTASLICE, name);
+
+                    Map<String, Object> allocations = (Map<String, Object>) ((Map<String, Object>) item.getAdditionalProperties().get("spec")).get("allocations");
+                    for (Map.Entry<String, Object> entry : allocations.entrySet()) {
+                        String key = entry.getKey();
+                        Map<String, Object> allocationMap = (Map<String, Object>) entry.getValue();
+                        if (allocationMap.get("namespace").toString().equalsIgnoreCase(workloadNamespace)
+                                && allocationMap.get("podName").toString().startsWith(workloadName)) {
+                            uuid.set(allocationMap.get("gpuUUID").toString());
+                        }
+                    }
+                });
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        } else {
+            LOGGER.info(AnalyzerConstants.AutoscalerConstants.InfoMsgs.NO_INSTASLICE_OBJECTS, namespace);
+        }
+        return uuid.get();
+    }
+
+    public String getMIGProfile(String workloadNamespace, String workloadName) {
+        if (!checkIfInstasliceExist())
+            return null;
+
+        AtomicReference<String> profile = new AtomicReference<>(null);
+
+        List<GenericKubernetesResource> instasliceObjects = getInstasliceObjects();
+        ObjectMapper mapper = new ObjectMapper();
+
+        if (null != instasliceObjects && !instasliceObjects.isEmpty()) {
+            try {
+                instasliceObjects.forEach(item -> {
+                    String name = item.getMetadata().getName();
+                    LOGGER.debug(AnalyzerConstants.AutoscalerConstants.InfoMsgs.FOUND_INSTASLICE, name);
+
+                    Map<String, Object> allocations = (Map<String, Object>) ((Map<String, Object>) item.getAdditionalProperties().get("spec")).get("allocations");
+                    for (Map.Entry<String, Object> entry : allocations.entrySet()) {
+                        String key = entry.getKey();
+                        Map<String, Object> allocationMap = (Map<String, Object>) entry.getValue();
+                        if (allocationMap.get("namespace").toString().equalsIgnoreCase(workloadNamespace)
+                                && allocationMap.get("podName").toString().startsWith(workloadName)) {
+                            profile.set(allocationMap.get("profile").toString());
+                            System.out.println(allocationMap.get("profile"));
```

Comment:
Is this needed?

Reply:
Thanks for pointing out, it's not needed. Have removed it.